python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -205,13 +247,26 @@ def dispatch_next_task(self) -> None:
                 return
         assert False, "Nothing to dispatch"
 
-    def get_output_blocking(self, output_split_idx: Optional[int]) -> MaybeRefBundle:
+    def get_output_blocking(self, output_split_idx: Optional[int]) -> RefBundle:
         """Get an item from this node's output queue, blocking as needed.
 
         Returns:
             The RefBundle from the output queue, or an error / end of stream indicator.
+
+        Raises:
+            StopIteration: If all outputs are already consumed.
+            Exception: If there was an exception raised during execution.
         """
         while True:
+            # Check if StreamingExecutor has caught an exception or is done execution.
+            if (
+                self.done_with_exception is not None
+                and not isinstance(self.done_with_exception, StopIteration)
+            ) or (
+                isinstance(self.done_with_exception, StopIteration)
+                and not self.outqueue
```

Comment:
what does `not self.outqueue` mean?

Reply:
It checks if the outqueue is empty, I've replaced it with something clearer.